### PR TITLE
feat(frontend): cache course descriptions

### PR DESCRIPTION
**Pull request by martanman**
before:

cfold

after: 

cfnew

### CODE REVIEW

File: frontend/src/components/CourseDescriptionPanel/CourseDescriptionPanel.tsx [hunk 1]
```diff
@@ -25,12 +25,23 @@ type CourseDescriptionPanelProps = {
   className?: string;
   courseCode: string;
   onCourseClick?: (code: string) => void;
+  courseResCache: React.MutableRefObject<CourseResCache>;
 };
 
+interface CourseInfo {
```
Comment (leonardo-fan) [inline frontend/src/components/CourseDescriptionPanel/CourseDescriptionPanel.tsx]: curious but why'd you chose to make this an interface over a type?

File: frontend/src/components/CourseDescriptionPanel/CourseDescriptionPanel.tsx [hunk 2]
```diff
@@ -25,12 +25,23 @@ type CourseDescriptionPanelProps = {
...
+  course?: Course;
+  pathFrom?: CourseList;
+  unlocked?: CoursesUnlockedWhenTaken;
+  courseCap?: EnrolmentCapacityData;
+}
+
+export type CourseResCache = Record<string, CourseInfo>;
```
Comment (leonardo-fan) [inline frontend/src/components/CourseDescriptionPanel/CourseDescriptionPanel.tsx]: nitpick: but can we make this `CourseDescInfoResCache` or something similar for specificity's sake

**Review by leonardo-fan — CHANGES_REQUESTED** [review]
awesome stuff Martin - nice improvement! lgtm but just a few stylistic/maintainability things - apart from the small comments, can you move the type definition/s to the `src/types` folder since it's used by multiple components.

File: frontend/src/components/CourseDescriptionPanel/CourseDescriptionPanel.tsx (hunk 1 above)
Comment (martanman) [reply frontend/src/components/CourseDescriptionPanel/CourseDescriptionPanel.tsx]: literally for no reason but now i see circles pretty much just uses types so ill change that

File: frontend/src/components/CourseDescriptionPanel/CourseDescriptionPanel.tsx (hunk 2 above)
Comment (martanman) [reply frontend/src/components/CourseDescriptionPanel/CourseDescriptionPanel.tsx]: aight

File: frontend/src/pages/CourseSelector/CourseSelector.tsx [hunk 3]
```diff
@@ -23,6 +24,7 @@ const CourseSelector = () => {
 
   const dispatch = useDispatch();
 
+  const courseResCache = useRef({} as CourseDescInfoResCache);
```
Comment (leonardo-fan) [inline frontend/src/pages/CourseSelector/CourseSelector.tsx]: nit can use more specific var name here too, but if u cbbs it's ok

File: frontend/src/pages/GraphicalSelector/GraphicalSelector.tsx [hunk 4]
```diff
@@ -12,6 +13,7 @@ const GraphicalSelector = () => {
   const [fullscreen, setFullscreen] = useState(false);
   const [courseCode, setCourseCode] = useState<string | null>(null);
   const [activeTab, setActiveTab] = useState(HELP_TAB);
+  const courseResCache = useRef({} as CourseDescInfoResCache);
```
Comment (leonardo-fan) [inline frontend/src/pages/GraphicalSelector/GraphicalSelector.tsx]: same as above

**Review by leonardo-fan — APPROVED** [review]
lgtm, can be merged - little nit comments if u want to address lel

File: frontend/src/pages/CourseSelector/CourseSelector.tsx (hunk 3 above)
Comment (martanman) [reply frontend/src/pages/CourseSelector/CourseSelector.tsx]: k

File: frontend/src/pages/GraphicalSelector/GraphicalSelector.tsx (hunk 4 above)
Comment (martanman) [reply frontend/src/pages/GraphicalSelector/GraphicalSelector.tsx]: yup

**Comment by lhjt** [comment]
@martanman do you plan to merge this anytime soon 👀

**Comment by martanman** [comment]
> @martanman do you plan to merge this anytime soon 👀

k yeah I will

**Pull request merged**